### DISCUSSION TITLE
gcamp_gazebo의 gcamp_world.launch.py에 대한 궁금증입니다.

### DISCUSSION THREAD

**Discussion by ckj0123** [Road-Balance/gcamp_ros2_basic#13] [Q&A] [answered]
ubuntu20.04에서 ros2 foxy를 사용중입니다.

![Screenshot from 2022-02-16 17-56-29](https://user-images.githubusercontent.com/95273933/154229979-f87ba054-8143-44b0-826f-8c492e89a27f.png)
![Screenshot from 2022-02-16 17-56-17](https://user-images.githubusercontent.com/95273933/154230103-3c75b5e4-b656-44cb-9916-20d496ad5d7d.png)
처음 gcamp_gazebo gcamp_world.launch.py를 실행시킨 gazebo와 rviz의 결과입니다.

![Screenshot from 2022-02-16 17-59-10](https://user-images.githubusercontent.com/95273933/154230388-e9176d64-33b8-4288-a7dd-9197bb824def.png)
![Screenshot from 2022-02-16 17-58-54](https://user-images.githubusercontent.com/95273933/154230412-b4fd9118-cc06-4169-a257-96e37b8e88c4.png)
그리고 두번째 gcamp_gazebo gcamp_world.launch.py를 실행시킨 gazebo와 rviz의 결과입니다.

첫번째 시도 당시에는  gazebo가 먼저 화면에 출력되고 그 이후에 rviz가 출력이 되었습니다.
이 결과로는, gazebo에 로봇이 출력되지 않고 rviz상으로도 센서데이터가 출력되지 않습니다.

하지만, 두번째 시도에서는 rviz가 먼저 출력되고 그 이후에 gazebo가 출력 되었습니다.
이 결과로는, gazebo와 rviz상 모두 로봇이 출력되는 결과가 나왔습니다.

순서와 상관이 있는것인지 궁금하며 순서와 관련이 있다면 rviz가 먼저 실행될 수 있도록 하는 방법이 있는지 궁금합니다.

감사합니다.

**Comment by kimsooyoung** [comment, marked as answer]
rviz는 로봇의 데이터를 시각화해주는 visualizer입니다. 로봇이 없다면 당연히 시각화할 데이터도 없기 때문에 아무런 변화가 없습니다.
따라서 gazebo 상의 로봇이 있어야 rviz를 통해 시각화가 가능합니다.

gazebo를 먼저 실행한 후 rviz를 실행하세요.
